restapi/src/main/java/io/stargate/web/resources/v2/schemas/UserDefinedTypesResource.java

```diff
@@ -0,0 +1,388 @@
+                    fieldName,
+                    Kind.Regular,
+                    Type.fromCqlDefinitionOf(keyspace, colDef.getTypeDefinition())));
+          }
+
+          UserDefinedType udt =
+              ImmutableUserDefinedType.builder()
+                  .keyspace(keyspaceName)
+                  .name(typeName)
+                  .addColumns(columns.toArray(new Column[0]))
+                  .build();
+
+          authenticatedDB
+              .getDataStore()
+              .queryBuilder()
+              .create()
+              .type(keyspaceName, udt)
+              .ifNotExists(udtAdd.getIfNotExists())
+              .build()
+              .execute(ConsistencyLevel.LOCAL_QUORUM)
+              .get();
+          return Response.status(Response.Status.CREATED)
+              .entity(Converters.writeResponse(Collections.singletonMap("name", typeName)))
+              .build();
+        });
+  }
+
+  @Timed
+  @DELETE
+  @ApiOperation(
+      value = "Delete a User Defined type (UDT)",
+      notes = "Delete a single user defined type (UDT) in the specified keyspace.")
+  @ApiResponses(
+      value = {
+        @ApiResponse(code = 204, message = "No Content"),
+        @ApiResponse(code = 401, message = "Unauthorized", response = Error.class),
+        @ApiResponse(code = 500, message = "Internal server error", response = Error.class)
+      })
+  @Path("/{typeName}")
+  public Response delete(
+      @ApiParam(
+              value =
+                  "The token returned from the authorization endpoint. Use this token in each request.",
+              required = true)
+          @HeaderParam(HEADER_TOKEN_AUTHENTICATION)
+          String token,
+      @ApiParam(value = "Name of the keyspace to use for the request.", required = true)
+          @PathParam("keyspaceName")
+          final String keyspaceName,
+      @ApiParam(
+              value = "Name of the user defined type (UDT) to use for the request.",
+              required = true)
+          @PathParam("typeName")
+          final String typeName,
+      @Context HttpServletRequest request) {
+    return RequestHandler.handle(
+        () -> {
+          AuthenticatedDB authenticatedDB = db.getDataStoreForToken(token, getAllHeaders(request));
+          Keyspace keyspace = authenticatedDB.getDataStore().schema().keyspace(keyspaceName);
+          if (keyspace == null) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                .entity(
+                    new Error(
+                        "keyspace does not exists", Response.Status.BAD_REQUEST.getStatusCode()))
+                .build();
+          }
+
+          db.getAuthorizationService()
+              .authorizeSchemaWrite(
+                  authenticatedDB.getAuthenticationSubject(),
+                  keyspaceName,
+                  null,
+                  Scope.DROP,
+                  SourceAPI.REST,
+                  ResourceKind.TYPE);
+
+          authenticatedDB
```

Comment:
Is it possible to catch the error when the type is still in use? That way we return a 4xx instead of 500?

Reply:
The 400 is already returned when there's an attempt to delete a in use UDT. I have added a test to track that.